src/errors/validationError.ts

```diff
@@ -0,0 +1,57 @@
+export enum ValidationErrorReason {
+    // Unfixable validation errors
+    UNSUPPORTED_TX_UPDATE = 'The transaction body entry update must not be included',
+    UNSUPPORTED_CERTIFICATE_GENESIS_KEY_DELEGATION = 'Certificate of type genesis_key_delegation is not supported and must not be included',
+    UNSUPPORTED_CERTIFICATE_MOVE_INSTANTANEOUS_REWARDS_CERT = 'Certificate of type move_instantaneous_rewards_cert is not supported and must not be included',
+    INTEGER_NOT_INT64 = 'Hardware wallets support integers up to int64, integers from -2^63 to 2^63-1',
+    UNSIGNED_INTEGER_NOT_UINT64 = 'Hardware wallets support unsigned integers up to uint64, unsigned integers from 0 to 2^64-1',
+    NUMBER_OF_ELEMENTS_EXCEEDS_UINT16 = 'The number of transaction elements individually must not exceed UINT16_MAX, i.e. 65535',
+    CERTIFICATES_MUST_HAVE_THE_SAME_TYPE_OF_STAKE_CREDENTIAL = 'All certificates included in a transaction must have the same type of stake credential',
+    WITHDRAWALS_MUST_HAVE_THE_SAME_TYPE_OF_STAKE_CREDENTIAL = 'All withdrawals included in a transaction must have the same type of stake credential',
+    CERTIFICATES_AND_WITHDRAWALS_STAKE_CREDENTIAL_TYPES_MUST_BE_CONSISTENT = 'The stake credential type of certificates must be consistent with the type used for withdrawals',
+    POOL_REGISTRATION_CERTIFICATE_WITH_OTHER_CERTIFICATES = 'If a transaction contains a pool registration certificate, then it must not contain any other certificate',
+    POOL_REGISTRATION_CERTIFICATE_WITH_WITHDRAWALS = 'If a transaction contains a pool registration certificate, then it must not contain any withdrawal',
+    POOL_REGISTRATION_CERTIFICATE_WITH_MINT_ENTRY = 'If a transaction contains a pool registration certificate, then it must not contain mint entry',
+
+    // Fixable valiadation errors
+    CBOR_IS_NOT_CANONICAL = 'CBOR is not canonical',
+    OPTIONAL_EMPTY_LISTS_AND_MAPS_MUST_NOT_BE_INCLUDED = 'Optional empty lists and maps must not be included as part of the transaction body or its elements',
+    OUTPUT_WITHOUT_TOKENS_MUST_BE_A_SIMPLE_TUPLE = 'Outputs containing no multi-asset tokens must be serialized as a simple tuple',
+}
+
+const FIXABLE = true
+const UNFIXABLE = false
+
+const validationErrorFixability: Record<ValidationErrorReason, boolean> = {
```

Comment:
The test is that the `Record` type guarantees that. I previously implemented it as a function with a switch statement basically just noting which `ValidationError`s are fixable:
```ts
const validationErrorFixability = (validationErrorReason) => {
  switch (validationErrorReason) {
  case ValidationErrorReason.CBOR_IS_NOT_CANONICAL:
  case ...:
    return true
  default:
    return false
  }
}
```
But I thought using the `Record` is better because if I add a new item to `ValidationErrorReason` TypeScript forces me to update the `validationErrorFixability` as well.